### PR TITLE
Add `parallel` option for cmake

### PR DESCRIPTION
**Pull request by chm-diederichs**

### CODE REVIEW

File: bin/bare-dev.js [hunk 1]
```diff
@@ -22,6 +22,11 @@ program
       .conflicts(['quiet'])
       .default(false)
   )
+  .addOption(
+    createOption('-j, --parallel <number>', 'parallelism')
+      .default(1)
+      .argParser(n => Number(n))
+  )
```
Comment (kasperisager) [inline bin/bare-dev.js:29]: This should be added to the `build` command, not the global command.

**Review by kasperisager — CHANGES_REQUESTED** [review]

File: bin/bare-dev.js (hunk 1 above)
Comment (chm-diederichs) [reply bin/bare-dev.js:29]: OK, will do. Shall I also add to the `install` command too?

Comment (kasperisager) [reply bin/bare-dev.js:29]: That'd be great, yep!

**Review by kasperisager — CHANGES_REQUESTED** [review]
As it currently stands, this will make all builds sequential by default as `--parallel 1` is passed to CMake unless something else is specified. Let's use the number of available cores as the default instead; might as well put them to use!

**Comment by kasperisager** [comment]
Perfect, that works as well!

**Review by kasperisager — APPROVED** [review]

**Comment by chm-diederichs** [comment]
> As it currently stands, this will make all builds sequential by default as `--parallel 1` is passed to CMake unless something else is specified. Let's use the number of available cores as the default instead; might as well put them to use!

I think by default cmake builds sequentially, maybe it is better to use `os.availableParallelism()` then, what do you think?

**Comment by mafintosh** [comment]
i think the normal saturation default is cores + 1 from back in my day

**Comment by kasperisager** [comment]
The default depends on the build system used. Ninja, for example, will be as parallel as possible unless otherwise specified, whereas MSBuild will be strictly sequential.

**Comment by chm-diederichs** [comment]
Ok, in which case seems appropriate to defer to the build system default

**Pull request merged**